src/shared/openssl-util.c

```diff
@@ -236,6 +236,91 @@ int openssl_hmac_many(
         return 0;
 }
 
+/* Perform Key-Based HMAC KDF. The mode must be "COUNTER" or "FEEDBACK". The parameter naming is from the
```

Comment:
Add an assert for 'mode' to enforce this condition, so that we don't get it wrong in the future

Reply:
ack